src/xopen/__init__.py

```diff
@@ -313,15 +321,20 @@ def close(self) -> None:
             return
         self.closed = True
         retcode = self.process.poll()
+        check_allowed_code_and_message = False
         if retcode is None:
             # still running
             self.process.terminate()
-            allow_sigterm = True
-        else:
-            allow_sigterm = False
-        self.process.wait()
+            check_allowed_code_and_message = True
+        try:
```

Comment:
I wonder if this is caused by https://github.com/pycompression/xopen/issues/16#issuecomment-5212402941